SpatialGDK/Source/SpatialGDK/Private/Utils/ComponentReader.cpp

```diff
@@ -108,7 +108,28 @@ void ComponentReader::ApplySchemaObject(Schema_Object* ComponentObject, UObject*
 
 	TArray<UProperty*> RepNotifies;
 
-	for (uint32 FieldId : UpdatedIds)
+	// If we are applying initial data, they must have come from a ComponentData (as it currently stands).
+	// ComponentData will be missing fields if they are completely empty (options, lists, and maps).
+	// However, we still want to apply this empty data, so we need to reconstruct the full
+	// list of field IDs for that component type (Data, OwnerOnly).
+	TArray<Schema_FieldId> InitialIds;
+	if (bIsInitialData)
+	{
+		const FClassInfo& ClassInfo = ClassInfoManager->GetClassInfoByComponentId(ComponentId);
+
+		int32 SchemaType = ClassInfoManager->GetCategoryByComponentId(ComponentId);
+		for (int32 HandleIndex = 0; HandleIndex < BaseHandleToCmdIndex.Num(); HandleIndex++)
+		{
+			const FRepParentCmd& Parent = Parents[Cmds[BaseHandleToCmdIndex[HandleIndex].CmdIndex].ParentIndex];
+			if (GetGroupFromCondition(Parent.Condition) == SchemaType)
+			{
+				InitialIds.Add(HandleIndex + 1);
+			}
+		}
+	}
+	const TArray<Schema_FieldId>& IdsToIterate = bIsInitialData ? InitialIds : UpdatedIds;
```

Comment:
So the `UpdateIds` will 100% not contain something we might miss in the in `InitialIds`?